packages/amplify-category-notifications/src/multi-env-manager.ts

```diff
@@ -1,28 +1,128 @@
+      // Only update notifications (amplifyMeta and backendConfig) with current pinpoint meta
+      if (pinpointNotificationsMeta.output?.Id) {
+        const { envName } = context.exeInfo.localEnvInfo;
+        const regulatedResourceName = PinpointName.extractResourceName(pinpointNotificationsMeta.Name, envName);
+        // update amplifyMeta.notifications from current pinpoint meta
+        context.exeInfo.amplifyMeta.notifications = {
+          [regulatedResourceName]: {
+            Id: pinpointNotificationsMeta.output.Id,
+            ResourceName: regulatedResourceName,
+            Name: pinpointNotificationsMeta.Name,
+            Region: pinpointNotificationsMeta.Region,
+            service: pinpointNotificationsMeta.service,
+            output: pinpointNotificationsMeta.output,
+            lastPushTimeStamp: pinpointNotificationsMeta.lastPushTimeStamp,
+          },
+        };
+        // update backendConfig with current pinpoint meta
+        const availableChannels = Notifications.ChannelCfg.getAvailableChannels();
+        const enabledChannels = availableChannels.filter(channelName => channelName in pinpointNotificationsMeta.output);
+        context.exeInfo.backendConfig.notifications = (context.exeInfo.backendConfig.notifications) || {};
+        context.exeInfo.backendConfig.notifications = {
+          [regulatedResourceName]: {
+            service: pinpointNotificationsMeta.service,
+            channels: enabledChannels,
+          },
+        };
+      }
+      // only save the state for pull
+      await writeData(context, undefined);
+    }
   }
   return pinpointNotificationsMeta;
 };
 
+const getAnalyticsResourcesFromMeta = (amplifyMeta: $TSMeta, supportedServiceName: string): IAnalyticsResource[] => {
+  const resourceList: IAnalyticsResource[] = [];
+  if (amplifyMeta[AmplifyCategories.ANALYTICS]) {
+    const categoryResources = amplifyMeta[AmplifyCategories.ANALYTICS];
+    Object.keys(categoryResources).forEach(resource => {
+      // if resourceProviderService is provided, then only return resources provided by that service
+      // else return all resources. e.g. Pinpoint, Kinesis
+      if (!supportedServiceName || categoryResources[resource].service === supportedServiceName) {
+        resourceList.push({
+          category: AmplifyCategories.ANALYTICS,
+          resourceName: resource,
+          service: categoryResources[resource].service,
+          region: categoryResources[resource]?.output?.Region,
+          id: categoryResources[resource]?.output?.Id,
+          output: categoryResources[resource]?.output,
+        });
+      }
+    });
+  }
+  return resourceList;
+};
+/**
+ * Utility function to check if amplify-cli is in pull mode or not.
+ * @param context amplify cli context
+ * @returns true if amplify-pull or env add/checkout commands are executed.
+ */
+export const isAmplifyCLIPulling = (context: $TSContext):boolean => (context?.input?.command)
```

Comment:
do we need to check for `env checkout` or `init --appId <id> --envName <name>`? Both of those commands will also do a pull under the hood

Reply:
after looking into the usage of this method decided to remvoe it as it was used only to prevent the spinner.fail calls in the catch blocks